pkgs/os-specific/linux/fuse/common.nix

```diff
@@ -43,50 +44,73 @@ in stdenv.mkDerivation rec {
     then [ meson ninja pkg-config ]
     else [ autoreconfHook gettext ];
 
-  outputs = [ "out" ] ++ lib.optional isFuse3 "common";
+  # The `common` output (`fuse-common` from top-level) is for `/etc` which shares
+  # between fuse2 and fuse3. From the documentation, `fuse3`'s common files are preferred,
+  # even for `fuse2`.
+  # https://github.com/libfuse/libfuse/blob/d372d3f80b5437e652ea501d8a4a917f7001b9d7/ChangeLog.rst#libfuse-300-2016-12-08
+  #
+  # It contains,
+  # - An init script `/etc/init.d/fuse3`.
+  #   This is unused for systemd-managed system.
+  #
+  # - A udev rules `/etc/udev/rules.d/99-fuse3.rules` to configure `/dev/fuse`.
+  #   This is included in systemd-udevd default rules since very early time, thus unneeded.
+  #   https://github.com/systemd/systemd/blob/0cf2dcf15402c60498165dbd3f14536766c05051/rules.d/50-udev-default.rules.in#L94
+  #
+  # - An example commented `/etc/fuse.conf`.
+  #   In NixOS, we control and generate it in `nixos/modules/programs/fuse.nix`.
+  #   Also the example actually enables nothing.
+  #
+  # Thus we really not need this output currently.
+  outputs = [ "bin" "out" "dev" "man" ] ++ lib.optional isFuse3 "common";
 
   mesonFlags = lib.optionals isFuse3 [
-    "-Dudevrulesdir=/udev/rules.d"
     "-Duseroot=false"
+    "-Dudevrulesdir=/udev/rules.d"
   ];
 
+  # Ensure that FUSE calls the setuid wrapper, not
+  # $out/bin/fusermount. It falls back to calling fusermount in
+  # $PATH, so it should also work on non-NixOS systems.
+  NIX_CFLAGS_COMPILE = ''-DFUSERMOUNT_DIR="/run/wrappers/bin"'';
+
   preConfigure = ''
-    export MOUNT_FUSE_PATH=$out/sbin
+    substituteInPlace lib/mount_util.c \
+      --replace "/bin/mount" "${lib.getBin util-linux}/bin/mount" \
+      --replace "/bin/umount" "${lib.getBin util-linux}/bin/umount"
```

Comment:
Is this also run as root? Otherwise the suid wrapper would be required.

Reply:
Yes. `fusermount{3,}` itself is an SUID binary, `mount.fuse{3,}` requires root.